contracts/GoodGhosting.sol

```diff
@@ -238,6 +238,20 @@ contract GoodGhosting is Ownable, Pausable {
         );
         // Decreases the totalGamePrincipal on earlyWithdraw
         totalGamePrincipal = totalGamePrincipal.sub(withdrawAmount);
+        // BUG FIX - Deposit External Pool Tx reverted after an early withdraw
+        // Fixed by first checking at what segment early withdraw happens if > 0 then re-assign current segment as -1
+        // Since in deposit external pool the amount is calculated from the segmentDeposit mapping
+        // and the amount is reduced by withdrawAmount
+        uint256 currentSegment = getCurrentSegment();
+        if (currentSegment > 0) {
+            currentSegment = currentSegment.sub(1);
+        }
+        if (segmentDeposit[currentSegment] > 0) {
+            segmentDeposit[currentSegment] = segmentDeposit[currentSegment].sub(
+            withdrawAmount
```

Comment:
This will revert if `segmentDeposit[currentSegment]` amount is less than `withdrawAmount`, which can happen.
In such case, `segmentDeposit[currentSegment]` should be set to zero instead, IMO.

Reply:
It'd be good put in place a unit test that covers both scenarios (withdraw amount less than and greater than `segmentDeposit[currentSegment]`)

Reply:
oh nice to point out

Reply:
we have the test case for greater amount will add one for less